src/render_helpers/mod.rs

```diff
@@ -233,49 +234,49 @@ pub fn render_to_vec(
     Ok(copy.to_vec())
 }
 
-#[cfg(feature = "xdp-gnome-screencast")]
 pub fn render_to_dmabuf(
     renderer: &mut GlesRenderer,
-    dmabuf: smithay::backend::allocator::dmabuf::Dmabuf,
+    dmabuf: Dmabuf,
     size: Size<i32, Physical>,
     scale: Scale<f64>,
     transform: Transform,
     elements: impl Iterator<Item = impl RenderElement<GlesRenderer>>,
 ) -> anyhow::Result<SyncPoint> {
     let _span = tracy_client::span!();
+    ensure!(
+        dmabuf.format().code == Fourcc::Xrgb8888
```

Comment:
This should not be here, render_to_dmabuf() can render to any format. This should be checked in screencopy upon receiving the buffer.

Reply:
I did it because that's what `render_to_shm` was and is doing.

Reply:
Because that one has to render and download in a specific pixel format for it to match up the data layout (and it's also only used for screencopy)